packages/studio-base/src/players/RosbridgePlayer.ts

```diff
@@ -476,14 +477,129 @@ export default class RosbridgePlayer implements Player {
       }
 
       const problemId = `message:${topicName}`;
-      topic.subscribe((message) => {
+
+      this.#rosClient.addTopicCallback(topicName, (message) => {
         if (!this.#providerTopics) {
           return;
         }
         try {
           const buffer = (message as { bytes: ArrayBuffer }).bytes;
           const bytes = new Uint8Array(buffer);
-          const innerMessage = messageReader.readMessage(bytes);
+          // const innerMessage = messageReader.readMessage(bytes);
+          if ( message._topic_type === 'sensor_msgs/msg/LaserScan' ) {
```

Comment:
this function got too long. Would it be possible to put the decoding for the different message types on independent functions so that we get a structure like this?

```
addTopicCallback
{
  if Laserscan
  {
    decode_laserscan()
  } 
  if PointCloud
  {
    decode_pointcloud()
  }
...
...
```

Instead of having all the decoding logic just placed in the function?

Reply:
troll haha ya le habia dicho eso